### PR TITLE
[RDBMS] Add support for advanced threat protection setting for a PostgreSQL flexible server

### PR DESCRIPTION
**Pull request by ambrahma**
**Related command**
`az postgres flexible-server advanced-threat-protection-setting update --resource-group testGroup --server-name testserver --state Enabled/Disabled' to enable/disable state for advanced threat protection setting.
`az postgres flexible-server advanced-threat-protection-setting show --resource-group testGroup --server-name testserver` to show the advanced server threat protection setting

**Description**
Added support for Advanced Server Threat Protection Setting for PostgreSQL flexible server. It supports commands to update and show the advanced threat protection setting

**Testing Guide**
Added automated tests

**History Notes**

[RDBMS] `az postgres flexible-server advanced-threat-protection-setting show`: Show advanced threat protection setting
[RDBMS] `az postgres flexible-server advanced-threat-protection-setting update`: Update advanced threat protection setting using `--state` as Enabled/Disabled

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 1]
```diff
@@ -809,6 +809,20 @@ def _flexible_server_params(command_group):
             c.argument('principal_type', options_list=['--type', '-t'], default='User', arg_type=get_enum_type(['User', 'Group', 'ServicePrincipal', 'Unknown']), help='Type of the Azure AD administrator.')
             c.argument('identity', help='Name or ID of identity used for AAD Authentication.', validator=validate_identity)
 
+        # server advanced threat protection settings
+        for scope in ['update', 'show']:
+            argument_context_string = '{} flexible-server advanced-threat-protection-setting {}'.format(command_group, scope)
+            with self.argument_context(argument_context_string) as c:
+                c.argument('resource_group_name', arg_type=resource_group_name_type)
```
Comment (Ran-br) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:816]: Is this only applicable in the context of the current subscription? (no subscription param)
Also, on az sql there's an option to specify '--ids' which enables some sort of bulk operations by calling show/ update on multiple resource ids: [az-sql-server-advanced-threat-protection-setting](https://learn.microsoft.com/en-us/cli/azure/sql/server/advanced-threat-protection-setting?view=azure-cli-latest#az-sql-server-advanced-threat-protection-setting-update:~:text=default%20value%3A%20False-,%2D%2Dids,-One%20or%20more)

Comment (ambrahma) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py:816]: Updated help

File: src/azure-cli/azure/cli/command_modules/rdbms/_helptext_pg.py [hunk 2]
```diff
@@ -746,3 +746,34 @@
...
 """
+
+helps['postgres flexible-server advanced-threat-protection-setting'] = """
+type: group
+short-summary: Manage advanced threat protection setting for a PostgreSQL flexible server.
+"""
+
+helps['postgres flexible-server advanced-threat-protection-setting update'] = """
+type: command
+short-summary: Updates advanced threat protection setting state for a flexible server.
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_helptext_pg.py:757]:
```suggestion
short-summary: Update advanced threat protection setting state for a flexible server.
```

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py [hunk 3]
```diff
@@ -184,3 +190,9 @@ def load_flexibleserver_command_table(self, _):
         g.custom_command('list', 'flexible_server_ad_admin_list')
         g.custom_show_command('show', 'flexible_server_ad_admin_show')
         g.custom_wait_command('wait', 'flexible_server_ad_admin_show')
+
+    with self.command_group('postgres flexible-server advanced-threat-protection-setting', postgres_flexible_server_threat_protection_settings_sdk,
+                            custom_command_type=flexible_servers_custom_postgres,
+                            client_factory=cf_postgres_flexible_server_threat_protection_settings) as g:
+        g.custom_command('show', 'flexible_server_threat_protection_get', custom_command_type=flexible_servers_custom_postgres)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py:197]:
```suggestion
        g.custom_show_command('show', 'flexible_server_threat_protection_get', custom_command_type=flexible_servers_custom_postgres)
```

**Comment by evelyn-ys** [comment]
Kind reminder that our code complete date is 11/28/2023 10:00 UTC. Please address comments and CI failures in time if you want to catch [November 2023 (2023-12-05)](https://github.com/Azure/azure-cli/milestone/136)

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py (hunk 3 above)
Comment (ambrahma) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py]: Thank you, this solved the issue.

**Comment by ambrahma** [comment]
> Kind reminder that our code complete date is 11/28/2023 10:00 UTC. Please address comments and CI failures in time if you want to catch [November 2023 (2023-12-05)](https://github.com/Azure/azure-cli/milestone/136)

I have addressed comments and CI failures. PR is ready for merge

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**